src/trackers/trackingParams.js

```diff
@@ -0,0 +1,374 @@
+
+        if (!siteData.initialUrl) {
+            return
+        }
+
+        const siteUrl = new URL(siteData.initialUrl)
+
+        if (!siteUrl.searchParams) {
+            return
+        }
+
+        // set the global crawlParams off of the first parameter string we see
+        if (!crawlParams.paramString) {
+            setCrawlParams(siteUrl)
+        }
+
+        processRequests(siteData, siteUrl)
+        processCookies(siteData, siteUrl)
+        
+        totalSites++
+    })
+
+
+    cleanupFinalData(results)
+    fs.writeFileSync(`${config.trackerDataLoc}/build-data/generated/tracking_parameters.json`, JSON.stringify({totalSites, params: results}, null, 4))
+
+}
+run()
+
+
+// Look though the site 'call.arguments' data, count first and third party cookies
+function processCookies (siteData, siteUrl) {
+    if (!(siteData.data && siteData.data.apis && siteData.data.apis.savedCalls)) {
+        return
+    }
+
+    siteData.data.apis.savedCalls.forEach(call => {
+        
+        if (!(call.arguments && call.arguments.length)) {
+            return
+        }
+
+        if (typeof call.arguments === 'string') {
+            call.arguments = JSON.parse(call.arguments)
+        }
+
+        // find matching param and args. Don't count args that match all parameters, these are just the full param strings
+        call.arguments.forEach(arg => {
+            const argMatches = []
+            for (const [param, value] of crawlParams.params.entries()) {
+                if (arg.match(value) && !hasFalsePositiveMatch(arg)) {
+                    argMatches.push({arg, param})
+                }
+            }
+
+            // skip args that matched all params
+            if (argMatches.length && argMatches.length !== crawlParams.params.length) {
+                argMatches.forEach(match => {
+                    countCookies(match.arg, siteUrl, match.param)
+                })
+            }
+        })
+    })
+}
+
+function processRequests (siteData, siteUrl) {
+    if (!siteData.data.requests) {
+        return
+    }
+
+    const siteOwner = _getEntity(siteUrl)
+
+    siteData.data.requests.forEach(req => {
+        let requestUrl
+        try {
+            requestUrl = new URL(req.url)
+        } catch (e) {
+            console.log(e)
+            return
+        }
+
+        let requestOwner  = _getEntity(requestUrl)
+
+        // skip first party and same entity
+        if (requestUrl.hostname === siteUrl.hostname || (siteOwner && (siteOwner === requestOwner))) {
+            return
+        }
+
+        // process request paths, skip any that match one of the false positive parameters
+        if (requestUrl.pathname && !hasFalsePositiveMatch(requestUrl.pathname)) {
+            for (const [fakeParamKey, fakeParamValue] of crawlParams.params.entries()) {
+                const pathMatch = requestUrl.pathname.match(fakeParamValue)
+                if (pathMatch) {
```

Comment:
This? https://github.com/duckduckgo/tracker-radar-detector/pull/37/files#diff-06ddea81a6543d5d566ac2e22faa02284e40b95510eafbacee144ebf7aefdedbR130

Reply:
We have this check in cookies where we make sure that not all param values appear in the cookie value (because if they are cookie is probably just storing whole url / search string), we don't seem to have similar check for requests - should we have it?